[jvm-packages] context is stopped in RabitTrackerHandler

I use scala tracker and run with spark using xgboost 0.81 version. get the error XGBoost train failed. The error stack is like in section 1 below.  The log info is like in section 2 below.
I found the reason is ml.dmlc.xgboost4j.scala.rabit.handler.RabitTrackerHandler class.
![image](https://user-images.githubusercontent.com/15651680/51021945-e8306a00-15bd-11e9-8328-6c32d49bf85e.png)
When I comment the 178 line, then rebuild xgboost4j, my test can run success.
Please check whether this is a defect.

1. error stack:
Caused by: ml.dmlc.xgboost4j.java.XGBoostError: XGBoostModel training failed
	at ml.dmlc.xgboost4j.scala.spark.XGBoost$.ml$dmlc$xgboost4j$scala$spark$XGBoost$$postTrackerReturnProcessing(XGBoost.scala:283)
	at ml.dmlc.xgboost4j.scala.spark.XGBoost$$anonfun$trainDistributed$4.apply(XGBoost.scala:240)
	at ml.dmlc.xgboost4j.scala.spark.XGBoost$$anonfun$trainDistributed$4.apply(XGBoost.scala:222)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.immutable.List.map(List.scala:285)
	at ml.dmlc.xgboost4j.scala.spark.XGBoost$.trainDistributed(XGBoost.scala:221)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:186)
	at ml.dmlc.xgboost4j.scala.spark.XGBoostRegressor.train(XGBoostRegressor.scala:48)
	at org.apache.spark.ml.Predictor.fit(Predictor.scala:118)

2. log info:
19/01/11 16:24:00 INFO XGBoostSpark: Rabit returns with exit code 3
[INFO] [01/11/2019 16:24:00.092] [RabitTracker-akka.actor.default-dispatcher-11] [akka://RabitTracker/user/Handler] Message [ml.dmlc.xgboost4j.scala.rabit.handler.RabitTrackerHandler$RequestCompletionFuture$] from Actor[akka://RabitTracker/deadLetters] to Actor[akka://RabitTracker/user/Handler#-1754393715] was not delivered. [1] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
19/01/11 16:24:00 ERROR XGBoost: XGBoostModel training failed

--- a/jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/rabit/handler/RabitTrackerHandler.scala
+++ b/jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/rabit/handler/RabitTrackerHandler.scala
@@ -175,7 +175,6 @@ private[scala] class RabitTrackerHandler(numWorkers: Int)
 
       if (shutdownWorkers.size == numWorkers) {
         promisedShutdownWorkers.success(shutdownWorkers.size)
-        context.stop(self)
       }
 
     case WorkerRecover(prevRank, worldSize, jobId) =>

